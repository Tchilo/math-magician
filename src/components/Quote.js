import React from 'react';

const Quote = () => (
  <div data-testid="paragraph-parent" className="quote">
    <p>
      Mathmatics is not about numbers, equations,
      computations, or algorithms:
      it is about understanding.-William Paul Thurston
    </p>
  </div>
);
export default Quote;
