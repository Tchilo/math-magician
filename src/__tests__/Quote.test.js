import { render, screen } from '@testing-library/dom';
import renderer from 'react-test-renderer';
import Quote from '../components/Quote';

describe('Tests Quote page', () => {
  test('Checking the existence of the paragraph', () => {
    render(<Quote />);
    const paragraphParent = screen.getByTestId('paragraph-parent');
    expect(paragraphParent).toContainHTML('P');
  });

  test('renders as expected', () => {
    const tree = renderer.create(<Quote />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
