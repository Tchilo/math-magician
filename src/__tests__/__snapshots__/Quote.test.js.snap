// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests Quote page renders as expected 1`] = `
<div
  className="quote"
  data-testid="paragraph-parent"
>
  <p>
    Mathmatics is not about numbers, equations, computations, or algorithms: it is about understanding.-William Paul Thurston
  </p>
</div>
`;
